A word about this patch. There is more places with compiling issues. May upstream updated clang?

```
[  0% 1/55606] /bin/bash -c "PWD=/proc/self/cwd /usr/bin/ccache prebuilts/clang/host/linux-x86/clang-4691093/bin/clang++ 	-I device/samsung/smdk4412-common/include -I device/samsung/i9300/include -I hardware/samsung/exynos4/hal/libfimg4x/../include -I hardware/samsung/exynos4/hal/libfimg4x -I /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/SHARED_LIBRARIES/libfimg_intermediates -I /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/gen/SHARED_LIBRARIES/libfimg_intermediates -I libnativehelper/include_jni \$(cat /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/SHARED_LIBRARIES/libfimg_intermediates/import_includes)  -I system/core/include -I system/media/audio/include -I hardware/libhardware/include -I hardware/libhardware_legacy/include -I hardware/ril/include -I libnativehelper/include -I frameworks/native/include -I frameworks/native/opengl/include -I frameworks/av/include -isystem /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/include -isystem bionic/libc/include -isystem bionic/libc/kernel/uapi -isystem bionic/libc/kernel/uapi/asm-arm -isystem bionic/libc/kernel/android/scsi -isystem bionic/libc/kernel/android/uapi -c  -fomit-frame-pointer -DANDROID -fmessage-length=0 -W -Wall -Wno-unused -Winit-self -Wpointer-arith -no-canonical-prefixes -DNDEBUG -UDEBUG -fno-exceptions -Wno-multichar -O2 -g -fno-strict-aliasing -fdebug-prefix-map=/proc/self/cwd= -D__compiler_offsetof=__builtin_offsetof -Werror=int-conversion -Wno-reserved-id-macro -Wno-format-pedantic -Wno-unused-command-line-argument -fcolor-diagnostics -Wno-expansion-to-defined -Wno-zero-as-null-pointer-constant -fdebug-prefix-map=\$PWD/= -ffunction-sections -fdata-sections -fno-short-enums -funwind-tables -fstack-protector-strong -Wa,--noexecstack -D_FORTIFY_SOURCE=2 -Wstrict-aliasing=2 -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Werror=date-time -Werror=format-security -nostdlibinc -msoft-float -march=armv7-a -mfloat-abi=softfp -mfpu=neon  -target arm-linux-androideabi -Bprebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9/arm-linux-androideabi/bin  -Wsign-promo -Wno-inconsistent-missing-override -Wno-null-dereference -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -Wno-thread-safety-negative -Wno-gnu-include-next -fvisibility-inlines-hidden  -std=gnu++14 -mthumb -Os  -fno-rtti -Wall -Werror -fPIC -D_USING_LIBCXX -DANDROID_STRICT   -Werror=int-to-pointer-cast -Werror=pointer-to-int-cast -Werror=address-of-temporary -Werror=return-type -Wno-tautological-constant-compare -Wno-null-pointer-arithmetic -Wno-enum-compare -Wno-enum-compare-switch   -MD -MF /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/SHARED_LIBRARIES/libfimg_intermediates/FimgExynos4.d -o /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/SHARED_LIBRARIES/libfimg_intermediates/FimgExynos4.o hardware/samsung/exynos4/hal/libfimg4x/FimgExynos4.cpp"
FAILED: /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/SHARED_LIBRARIES/libfimg_intermediates/FimgExynos4.o 
/bin/bash -c "PWD=/proc/self/cwd /usr/bin/ccache prebuilts/clang/host/linux-x86/clang-4691093/bin/clang++ 	-I device/samsung/smdk4412-common/include -I device/samsung/i9300/include -I hardware/samsung/exynos4/hal/libfimg4x/../include -I hardware/samsung/exynos4/hal/libfimg4x -I /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/SHARED_LIBRARIES/libfimg_intermediates -I /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/gen/SHARED_LIBRARIES/libfimg_intermediates -I libnativehelper/include_jni \$(cat /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/SHARED_LIBRARIES/libfimg_intermediates/import_includes)  -I system/core/include -I system/media/audio/include -I hardware/libhardware/include -I hardware/libhardware_legacy/include -I hardware/ril/include -I libnativehelper/include -I frameworks/native/include -I frameworks/native/opengl/include -I frameworks/av/include -isystem /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/include -isystem bionic/libc/include -isystem bionic/libc/kernel/uapi -isystem bionic/libc/kernel/uapi/asm-arm -isystem bionic/libc/kernel/android/scsi -isystem bionic/libc/kernel/android/uapi -c  -fomit-frame-pointer -DANDROID -fmessage-length=0 -W -Wall -Wno-unused -Winit-self -Wpointer-arith -no-canonical-prefixes -DNDEBUG -UDEBUG -fno-exceptions -Wno-multichar -O2 -g -fno-strict-aliasing -fdebug-prefix-map=/proc/self/cwd= -D__compiler_offsetof=__builtin_offsetof -Werror=int-conversion -Wno-reserved-id-macro -Wno-format-pedantic -Wno-unused-command-line-argument -fcolor-diagnostics -Wno-expansion-to-defined -Wno-zero-as-null-pointer-constant -fdebug-prefix-map=\$PWD/= -ffunction-sections -fdata-sections -fno-short-enums -funwind-tables -fstack-protector-strong -Wa,--noexecstack -D_FORTIFY_SOURCE=2 -Wstrict-aliasing=2 -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Werror=date-time -Werror=format-security -nostdlibinc -msoft-float -march=armv7-a -mfloat-abi=softfp -mfpu=neon  -target arm-linux-androideabi -Bprebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9/arm-linux-androideabi/bin  -Wsign-promo -Wno-inconsistent-missing-override -Wno-null-dereference -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -Wno-thread-safety-negative -Wno-gnu-include-next -fvisibility-inlines-hidden  -std=gnu++14 -mthumb -Os  -fno-rtti -Wall -Werror -fPIC -D_USING_LIBCXX -DANDROID_STRICT   -Werror=int-to-pointer-cast -Werror=pointer-to-int-cast -Werror=address-of-temporary -Werror=return-type -Wno-tautological-constant-compare -Wno-null-pointer-arithmetic -Wno-enum-compare -Wno-enum-compare-switch   -MD -MF /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/SHARED_LIBRARIES/libfimg_intermediates/FimgExynos4.d -o /run/media/leonardo/Storage/leonardo/AndroidDev/i9300/Chromos_LOS16/out/target/product/i9300/obj/SHARED_LIBRARIES/libfimg_intermediates/FimgExynos4.o hardware/samsung/exynos4/hal/libfimg4x/FimgExynos4.cpp"
In file included from hardware/samsung/exynos4/hal/libfimg4x/FimgExynos4.cpp:24:
In file included from hardware/samsung/exynos4/hal/libfimg4x/FimgExynos4.h:41:
hardware/samsung/exynos4/hal/libfimg4x/../include/FimgApi.h:114:32: error: unused parameter 'rhs' [-Werror,-Wunused-parameter]
        FimgApi(const FimgApi& rhs) {}
                               ^
hardware/samsung/exynos4/hal/libfimg4x/../include/FimgApi.h:138:1: error: struct 'FimgApi' was previously declared as a class [-Werror,-Wmismatched-tags]
struct FimgApi *createFimgApi();
^
hardware/samsung/exynos4/hal/libfimg4x/../include/FimgApi.h:103:7: note: previous use is here
class FimgApi
      ^
hardware/samsung/exynos4/hal/libfimg4x/../include/FimgApi.h:138:1: note: did you mean class here?
struct FimgApi *createFimgApi();
^~~~~~
class
hardware/samsung/exynos4/hal/libfimg4x/FimgExynos4.cpp:222:12: error: unused variable 'mmap_base' [-Werror,-Wunused-variable]
    void * mmap_base;
           ^
hardware/samsung/exynos4/hal/libfimg4x/FimgExynos4.cpp:323:12: error: struct 'FimgApi' was previously declared as a class [-Werror,-Wmismatched-tags]
extern "C" struct FimgApi * createFimgApi()
           ^
hardware/samsung/exynos4/hal/libfimg4x/../include/FimgApi.h:103:7: note: previous use is here
class FimgApi
      ^
hardware/samsung/exynos4/hal/libfimg4x/FimgExynos4.cpp:323:12: note: did you mean class here?
extern "C" struct FimgApi * createFimgApi()
           ^~~~~~
           class
hardware/samsung/exynos4/hal/libfimg4x/FimgExynos4.cpp:333:42: error: unused parameter 'ptrFimgApi' [-Werror,-Wunused-parameter]
extern "C" void destroyFimgApi(FimgApi * ptrFimgApi)
                                         ^
5 errors generated.
ninja: build stopped: subcommand failed.
21:12:54 ninja failed with: exit status 1
21:12:54 Ninja log not modified, not importing any entries.

#### failed to build some targets (10 seconds) ####
```

Just a comment: For this issue I'm using this patch in android_local_manifests_i9300:

```
diff --git a/fetch.sh b/fetch.sh
index a4e3d49..c235770 100755
--- a/fetch.sh
+++ b/fetch.sh
@@ -13,8 +13,8 @@ fetch() {
                proj=$(echo $line | cut -d " " -f1)
                echo -e $CL_BLU"Fetching patches for $proj"$CL_RST
                echo
-               echo "git -C $ANDROID_BUILD_TOP/$proj fetch https://github.com/CustomROMS/$(path2name $proj) lineage-15.1"
-               git -C $ANDROID_BUILD_TOP/$proj fetch https://github.com/CustomROMS/$(path2name $proj) lineage-15.1
+               echo "git -C $ANDROID_BUILD_TOP/$proj fetch https://github.com/CustomROMS/$(path2name $proj) lineage-16.0-i9300"
+               git -C $ANDROID_BUILD_TOP/$proj fetch https://github.com/CustomROMS/$(path2name $proj) lineage-16.0-i9300
                echo ""; echo "";
        done < $ANDROID_BUILD_TOP/android_local_manifests_i9300/patches.txt
 }
```

--- a/exynos4/hal/include/FimgApi.h
+++ b/exynos4/hal/include/FimgApi.h
@@ -18,6 +18,10 @@
 **
 */
 
+#pragma clang diagnostic ignored "-Wunused-parameter"
+#pragma clang diagnostic ignored "-Wmismatched-tags"
+#pragma clang diagnostic ignored "-Wunused-variable"
+
 #ifndef FIMG_API_H
 #define FIMG_API_H
 

--- a/exynos4/hal/libfimg4x/FimgApi.cpp
+++ b/exynos4/hal/libfimg4x/FimgApi.cpp
@@ -17,6 +17,11 @@
 **
 */
 
+#pragma clang diagnostic ignored "-Wsign-compare"
+#pragma clang diagnostic ignored "-Wformat"
+/* C++11 violation */
+#pragma clang diagnostic ignored "-Wwritable-strings"
+
 #define LOG_NDEBUG 0
 #define LOG_TAG "SKIA"
 #include <utils/Log.h>

--- a/exynos4/hal/libfimg4x/FimgExynos4.cpp
+++ b/exynos4/hal/libfimg4x/FimgExynos4.cpp
@@ -17,6 +17,10 @@
 **
 */
 
+#pragma clang diagnostic ignored "-Wunused-parameter"
+#pragma clang diagnostic ignored "-Wmismatched-tags"
+#pragma clang diagnostic ignored "-Wunused-variable"
+
 #define LOG_NDEBUG 0
 #define LOG_TAG "FimgExynos4"
 #include <utils/Log.h>

--- a/exynos4/hal/libhwjpeg/jpeg_hal_unit.cpp
+++ b/exynos4/hal/libhwjpeg/jpeg_hal_unit.cpp
@@ -16,6 +16,8 @@
 
 #define LOG_TAG "libhwjpeg"
 
+#pragma clang diagnostic ignored "-Wunused-function"
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/types.h>
@@ -34,8 +36,6 @@
 #include <log/log.h>
 
 #include "jpeg_hal.h"
-
-#pragma clang diagnostic ignored "-Wunused-function"
 
 #ifdef JPEG_PERF_MEAS
 unsigned long measure_time(struct timeval *start, struct timeval *stop)

